sdk/identity/identity/src/credentials/managedIdentityCredential/imdsMsi.ts

```diff
@@ -129,11 +143,28 @@ export const imdsMsi: MSI = {
       `Using the IMDS endpoint coming form the environment variable MSI_ENDPOINT=${process.env.MSI_ENDPOINT}, and using the cloud shell to proceed with the authentication.`
     );
 
-    return msiGenericGetToken(
-      identityClient,
-      prepareRequestOptions(resource, clientId),
-      expiresInParser,
-      getTokenOptions
+    let nextDelayInMs = imdsMsiRetryConfig.startDelayInMs;
+    for (let retries = 0; retries < imdsMsiRetryConfig.maxRetries; retries++) {
```

Comment:
How does the retry policy set for the client affect this?

Where are we disabling the retry policy of the client?

Reply:
The retry policy in core-http will never catch this since 404 is not a special case in core. In this code, only if the response has a status code of 404, we retry. This never interferes with the retry policy.